src/nvim/screen.c

```diff
@@ -4548,7 +4548,7 @@ static void screen_line(int row, int coloff, int endcol, int clear_width, int rl
       int c;
 
       c = fillchar_vsep(&hl);
-      if (ScreenLines[off_to] != c
+      if (ScreenLines[off_to] != (schar_T)c
```

Comment:
This will be included in #1657, so the vim-patch commit should only update the version.c.